program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,88 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+import logging
+
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+from program_intent_engagement.apps.core.models import ProgramIntent, User
+
+log = logging.getLogger(__name__)
+
+
+class ProgramIntentsView(APIView):
+    # todo: add expected program intent data
+    """
+    View to insert program intents.
+
+    Given a program intent, this view will either insert the program intent. If the given program intent already
+    exists, no action is taken (no duplicate row).
+
+    Path: /api/[version]/program_intents
+
+    Accepts: [POST]
+
+    HTTP POST
+    Inserts a Program Intent.
+    Expected POST data:
+    POST data parameters:
+     * program_uuid: The uuid of the program intent.
+     * reason: The reason for how intent was measured.
+     * certainty: The certainty of the intent for the user.
+     * effective_timestamp: When the event occurred that created the intent.
+    Returns:
+     * 200: The program intent was inserted.
+     * 400: The program intent request was invalid.
+     * 409: Duplicate program intent was not inserted.
```

Comment:
@MichaelRoytman good call! Since we're now using the `IsAuthenticated` permission class, should I no longer include the 403 response code in the docstring? I guess that's assuming that that class (`IsAuthenticated`) has the responsibility for the 403 response code.